.github/workflows/test.yml

```diff
@@ -0,0 +1,35 @@
+# GitHub Action to compile and test
+name: test
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        task: [data-check, codespell]
+        #spellintian, nosetests, karma, lint, closure-compiler, 
+    steps:
+      - uses: actions/checkout@master
+      - name: Install dependencies
+        env:
+          TASK: ${{ matrix.task }}
+        run: |
+          if [ "$TASK" = "nosetests" -o "$TASK" = "flake8" -o "$TASK" = "flake8-wip" ]; then pip install json-spec; fi
+          if [ "$TASK" = "karma" -o "$TASK" = "lint" -o "$TASK" = "closure-compiler" ]; then npm install -g grunt-cli; fi
+          if [ "$TASK" = "karma" -o "$TASK" = "lint" -o "$TASK" = "closure-compiler" ]; then npm install; fi
+          if [ "$TASK" = "flake8" -o "$TASK" = "flake8-wip" ]; then pip install flake8; fi
+          # If this causes SSL errors, then Sourceforge is probably in disaster recovery mode and needing Javascript. Could switch to a specific mirror, e.g. kent.dl.sourceforge.net
+          if [ "$TASK" = "pychecker" -o "$TASK" = "pychecker-wip" ]; then pip install http://sourceforge.net/projects/pychecker/files/pychecker/0.8.19/pychecker-0.8.19.tar.gz/download; fi
+          #if [ "$TASK" = "codespell" ]; then pip install git+https://github.com/codespell-project/codespell.git; fi
+          if [ "$TASK" = "codespell" ]; then pip install codespell; fi
+          sudo apt-get install xvfb
```

Comment:
Why not have those four `apt install` lines merged into one line with all packages installed in one go?

Reply:
Yeah that's a fair point. I think I was running them individually as the presence of a package already being installed broke the whole build!

Reply:
See also #304